packages/web/docs/src/components/slider.tsx

```diff
@@ -0,0 +1,124 @@
+      }}
+    >
+      <input
+        type="range"
+        min={0}
+        max={100}
+        step={0.1}
+        // for Safari
+        tabIndex={0}
+        className="h-10 w-full opacity-0 hover:cursor-grab active:cursor-grabbing [&::webkit-slider-thumb]:size-8"
+        {...rest}
+      />
+
+      <div // fill
+        className="pointer-events-none absolute top-[calc(50%-4px)] h-2 w-full rounded-lg bg-green-100 [clip-path:inset(0_0_0_0_round_8px)] after:absolute after:inset-0 after:left-[-120%] after:w-[120%] after:translate-x-[calc(var(--val)*1cqi)] after:bg-green-300"
+      />
+
+      <div // indicator
+        className={cn(
+          'after:text-green-1000 pointer-events-none absolute left-0 top-0 flex size-10 translate-x-[calc(var(--val)*(100cqi-100%)/100)] select-none items-center justify-center rounded-full bg-gradient-to-br from-green-500 to-green-700 text-center after:pointer-events-auto after:absolute after:top-[calc(-100%+3px)] after:whitespace-nowrap after:rounded-full after:bg-green-200 after:px-3 after:py-1 after:font-medium',
+          counter,
+        )}
+      >
+        <svg
+          xmlns="http://www.w3.org/2000/svg"
+          width="12"
+          height="16"
+          viewBox="0 0 12 16"
+          fill="currentColor"
+          className="absolute -top-full translate-y-[23px] text-green-200"
+        >
+          <path d="M0 8L6 0L12 8L6 16L0 8Z" />
+        </svg>
+        <img src={svgHref} width={24} height={24} alt="" />
+      </div>
+
+      <style jsx>{`
+        @property --val {
+          syntax: '<integer>';
+          initial-value: 0;
+          inherits: true;
+        }
+        @supports (animation-timeline: scroll()) and (animation-range: 0 100%) {
+          :has(> input) {
+            timeline-scope: --thumb;
+            animation: progress both linear;
+            animation-direction: reverse;
+            animation-timeline: --thumb;
+            animation-range: contain;
+          }
+          @keyframes progress {
+            to {
+              --val: 100;
+            }
+          }
+          input {
+            overflow: hidden;
+          }
+          ::-webkit-slider-thumb {
+            view-timeline: --thumb inline;
+          }
+        }
+
+        div,
+        div:after {
+          transition: transform var(--ease) 500ms;
+        }
+      `}</style>
+    </div>
+  );
+}
+
+/**
+ * We're using CSS scroll-driven animations to make the animation
+ * smoother. This is a polyfill for browsers that don't support it.
+ * https://caniuse.com/?search=scroll()
+ */
+function polyfillSlider(element: HTMLElement, cssProperty: `--${string}`) {
+  if (CSS.supports('(animation-timeline: view()) and (animation-range: 0 100%)')) {
+    return;
+  }
+
+  const input = element.querySelector('[type=range]') as HTMLInputElement;
+
+  const sync = () => {
+    const val = (Number(input.value) - Number(input.min)) / (Number(input.max) - Number(input.min));
```

Comment:
is this same input element?
```suggestion
  const sync = (event) => {
    const input = event.target
    const val = (Number(input.value) - Number(input.min)) / (Number(input.max) - Number(input.min));
```